### PR TITLE
[Dynamic Partition] Use ZonedDateTime to support set timezone

### PR DESCRIPTION
**Pull request by WingsGo**
This CL mainly support timezone in dynamic partition:
1. use new Java Time API to replace Calender.
2. support set time zone in dynamic partition parameters.

### CODE REVIEW

File: docs/en/administrator-guide/dynamic-partition.md [hunk 1]
```diff
@@ -89,6 +89,10 @@ The rules of dynamic partition are prefixed with `dynamic_partition.`:
 
     When specified as `MONTH`, the suffix format of the dynamically created partition name is `yyyyMM`, for example, `202003`.
 
+* `dynamic_partition.time_zone`
+
+    The time zone of the dynamic partition, if not filled in, defaults to the time zone of the current machine's system, such as `Asia/Shanghai`, if you want to know the supported TimeZone, you can run `TimeZone.getAvailableIDs()` in Java.
```
Comment (morningman) [inline docs/en/administrator-guide/dynamic-partition.md:94]: How about finding timezone here?
https://en.wikipedia.org/wiki/List_of_tz_database_time_zones

File: fe/src/main/java/org/apache/doris/catalog/DynamicPartitionProperty.java [hunk 2]
```diff
@@ -46,15 +48,14 @@
...
-    // TODO: support setting timezone.
     private TimeZone tz = TimeUtils.getDefaultTimeZone();
 
-
     public DynamicPartitionProperty(Map<String, String> properties) {
         if (properties != null && !properties.isEmpty()) {
             this.exist = true;
             this.enable = Boolean.parseBoolean(properties.get(ENABLE));
             this.timeUnit = properties.get(TIME_UNIT);
+            this.tz = TimeZone.getTimeZone(properties.getOrDefault(TIME_ZONE, ZoneId.systemDefault().toString()));
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/catalog/DynamicPartitionProperty.java:58]: Better move the time related functions to TimeUtils class

File: fe/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java [hunk 3]
```diff
@@ -148,11 +151,25 @@ private static void checkStartDayOfWeek(String val) throws DdlException {
         }
     }
 
+    private static void checkTimeZone(String val) throws DdlException {
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java:154]: Why not using `TimeUtils.checkTimeZoneValidAndStandardize()`?

Comment (WingsGo) [reply fe/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java:154]: In fact, i don't konw the function...

File: docs/en/administrator-guide/dynamic-partition.md (hunk 1 above)
Comment (WingsGo) [reply docs/en/administrator-guide/dynamic-partition.md:94]: done

File: fe/src/main/java/org/apache/doris/catalog/DynamicPartitionProperty.java (hunk 2 above)
Comment (WingsGo) [reply fe/src/main/java/org/apache/doris/catalog/DynamicPartitionProperty.java]: done

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**